WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -423,6 +423,65 @@ class PageListViewController: AbstractPostListViewController, UIViewControllerRe
         }
     }
 
+    func restorePost(_ apost: AbstractPost, at indexPath: IndexPath?) {
+        WPAnalytics.track(.postListRestoreAction, withProperties: propertiesForAnalytics())
+
+        // if the post was recently deleted, update the status helper and reload the cell to display a spinner
+        let postObjectID = apost.objectID
+
+        if let index = recentlyTrashedPostObjectIDs.index(of: postObjectID) {
+            recentlyTrashedPostObjectIDs.remove(at: index)
+        }
+
+        let postService = PostService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        postService.restore(apost, success: { [weak self] in
+            guard let strongSelf = self else {
+                return
+            }
+
+            var apost: AbstractPost
+
+            // Make sure the post still exists.
+            do {
+                apost = try strongSelf.managedObjectContext().existingObject(with: postObjectID) as! AbstractPost
+            } catch {
+                DDLogError("\(error)")
+                return
+            }
+
+            DispatchQueue.main.async {
```

Comment:
Rather than copy this entire method verbatim from `AbstractPostListViewController` just to add this one block, how about we instead modify the original method to accept an optional indexPath argument defaulted to nil, and then either conditionally execute this block there, or, if you don't want to extend `WPTableViewHandler` with a `refreshTableViewAt` method, call a new convenience method that `PageListViewController` can implement with this block (the default implementation could be a no-op.) 

I'm open to other options.  We should just try to avoid duplicating code where we can.

Reply:
I thought the same but i tried to avoid any change on the Abstract VC. The block solution is the one i prefer. Thanks for the hint 😉